lexos/static/js/scripts_upload.js

```diff
@@ -1,197 +1,197 @@
 $(function () {
-  if ($('.fa-folder-open-o')[0].style.display != 'none' &&
-    $('.fa-folder-open-o')[0].innerText.indexOf('Response') != -1) {
-    alert('Steven\'s special bug detected! Click OK and the page will load.')
-    $('.fa-folder-open-o')[0].style.display = 'none'
-  }
-
+  /*message for when you hover on the open folder icon on the top right*/
   $('[data-toggle="tooltip"]').tooltip()
-
   $('#uploadbrowse').click(function () {
     $('#fileselect').click()
   })
 
-  // ------------------- FILEDRAG -----------------------------
-
-  var allowedFileTypes = ['txt', 'xml', 'html', 'sgml', 'lexos']
-
+  //alias function
   function $id(id) {
     return document.getElementById(id)
   }
 
-  function AllowedFileType(filename) {
-    var splitName = filename.split('.')
-    var fileType = splitName[splitName.length - 1]
-    if ($.inArray(fileType, allowedFileTypes) > -1) {
-      return true
-    } else {
-      return false
-    }
-  }
-
-  // file drag hover
-  function FileDragHover(e) {
-    e.stopPropagation()
-    e.preventDefault()
-    e.target.className = (e.type == 'dragover' ? 'hover' : '')
-  }
-
-  var numberOfFileDone = parseInt($('.fa-folder-open-o')[0].id)
-
+  let numberOfFileDone = parseInt($('.fa-folder-open-o')[0].id)
   // file selection
   function FileSelectHandler(e) {
     // cancel event and hover styling
-
-    FileDragHover(e)
-
+    fileDragHover(e)
     // fetch FileList object
-    var files = e.target.files || e.dataTransfer.files
-
-    var totalFiles = files.length
-
-    // Make process bar back to 0
-    $('#progress-bar').html('').css({ 'width': '0px' })
-    $('#progress-bar').show()
-    $('#status').css('z-index', 50000)
-    $('#status').show()
-
-    // process all File objects
-    for (var i = 0, f; f = files[i]; i++) {
-      var added = 0
-
+    let files = e.target.files || e.dataTransfer.files
+    // total number of files uploaded
+    let totalFiles = files.length
+    //resets the progress bar to defualt message after a file upload
+    resetProgressBar()
+
+    const progress = $('#progress'),
+          progressBar = $('#progress-bar')
+
+    //Process all File objects
+    for (let i = 0, f; f = files[i]; i++) {
```

Comment:
`for (const f of files)`, https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...of